### DISCUSSION TITLE
📁 ディレクトリ構成どうしますか？

### DISCUSSION THREAD

**Discussion by kagomen** [lgtm-factory/lgtm-factory#71] [Q&A]
- 機能ベース
- 階層型
- DDD
- アトミックデザイン
なんか色々あるんですね……👀

https://www.aki--dev.com/entry/2024/05/22/085500

**Comment by kagomen** [comment]
この動画で紹介されてる方法、開発しやすそう！
でも、共有するの大変かな……
あまり気にしないで開発した方が良いんですかね……？

https://www.youtube.com/watch?v=ekUQ043k2TQ

**Reply by kazzyfrog** [reply]
機能ベースでfeatureで管理するのも割と一般的なので、
練習がてらそれでやってみてもいいかもですね！

今回そこまで機能ないので、やり過ぎ感はあるかもですが！w

**Reply by kagomen** [reply]
なるほどー！
たしかに、冷静に考えると機能少ないのかー……

自分で開発してるアプリで雑にフォルダ分けしたことをめちゃくちゃ後悔してるので、そうなりたくないなーって思ったのですが、そもそもLGTMはSPAっぽい感じですもんね……！

**Reply by kazzyfrog** [reply]
まあでも、後々の拡張可能性とか考えたら、
どうなるかはまだ分からないけど、そこそこしっかり目にディレクトリ分割するケースもありますよ！

結局、最小の構成ならどんなパターン構成でも、そこまで困らないと思うので、
規模が大きくなってきた時のため？のディレクトリ戦略だと思うので！👍

**Reply by kazzyfrog** [reply]
あ、でも、よく考えたら、APIのエンドポイントに関するドキュメント（使用法・注意事項）とか、
簡単なページ作成してもいいのかも。

GitHub上のREADMEとかに書くだけでもいいけど、
意外とこだわろうと思えば、割と分厚くもできそう！

こうトリビュートしたい人向けのドキュメントも、サイトに掲載しても良いかも！

**Reply by kazzyfrog** [reply]
もっと言えば、各デザインの個別ページ？とか作っても面白そうだなと！

作った人が、簡潔に書く感じ

画像生成とは別で、みたい人が見る感じの詳細画面的な

**Reply by kagomen** [reply]
Swagger、使ってみたい……！！！！

**Reply by kazzyfrog** [reply]
ありですね！
Next.jsで、いい感じにOAS生成するライブラリとかありそう。。！

**Reply by kagomen** [reply]
Swagger以外にも色々あるんですね👀

**Reply by kazzyfrog** [reply]
あ、まぎらわしかったですね！

Swaggerが、OpenAPI仕様（OAS）っていう名称に変わった感じです！
同じものを指している認識でOKです！👍
